src/Components/LifecycleFilters/LifecycleFilters.tsx

```diff
@@ -238,11 +227,51 @@ export const LifecycleFilters: React.FunctionComponent<LifecycleFiltersProps> =
 
   const isSystemsView = lifecycleDropdownValue === RHEL_SYSTEMS_DROPDOWN_VALUE;
 
+  // Define filter controls as variables to ensure they're always valid React elements
+  const nameSearchInput = (
+    <SearchInput
+      placeholder="Filter by name"
+      value={nameFilter}
+      onChange={(_event, value) => setNameFilter(value)}
+      onClear={() => setNameFilter('')}
+      aria-label="Filter by name"
+    />
+  );
+
+  const rhelVersionSelect = (
+    <Select
+      aria-label="RHEL version"
+      isOpen={isRhelSelectOpen}
+      onOpenChange={(open) => setIsRhelSelectOpen(open)}
+      onSelect={onRhelSelect}
+      selected={selectedRhelVersions}
+      toggle={(toggleRef: React.Ref<HTMLDivElement>) => (
+        <MenuToggle
+          ref={toggleRef}
+          onClick={() => setIsRhelSelectOpen((p) => !p)}
+          isExpanded={isRhelSelectOpen}
+        >
+          RHEL versions
+          {selectedRhelVersions.length > 0 && <Badge isRead>{selectedRhelVersions.length}</Badge>}
+        </MenuToggle>
```

Comment:
```suggestion
        <MenuToggle ref={toggleRef} onClick={() => setIsRhelSelectOpen((p) => !p)} isExpanded={isRhelSelectOpen}>
          RHEL versions
          {selectedRhelVersions.length > 0 && <Badge isRead>{selectedRhelVersions.length}</Badge>}
        </MenuToggle>
```
just some formatting